pkg/pillar/zedcloud/send.go

```diff
@@ -766,7 +773,38 @@ func SendOnIntf(workContext context.Context, ctx *ZedCloudContext, destURL strin
 				LocalAddr: &localTCPAddr,
 				Timeout:   clientConfig.TCPHandshakeTimeout,
 			}
-			transport.DialContext = d.DialContext
+			resolverCache := ctx.ResolverCacheFunc
+			dialWithResolverCache := func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
Can we not declare this as a function at global scope? Avoids accidentally using variables from this functions scope.

Reply:
I moved these nested functions out into the global scope. It really makes the code more readable and a bit less bug prone.